### ISSUE TITLE
WYSIWYG editor does not properly report cursor position to the at.who @mention plugin

### ISSUE THREAD

**Issue by charrondev** [vanilla/vanilla#6829]
This can break mentions. This happens both with newlines and with unicode characters.

These issues are rooted in the `EditableCaret.prototype.getPos`  function in `jquery.atwho.js`.

Both of the following contents represent issues. `[CURSOR]` represents your cursor position.

## Case 1
```
dlasdlf @"Joe0jasdfjsdaj39933íöéé"[CURSOR] dfssdfaasfd
```

You must press backspace multiple times before the the @mention is triggered again.

## Case 2
```
@line1

[CURSOR]

Text
```

Press enter. The mention popup will attempt to appear briefly.

**Comment by chrisvanpatten** [comment]
I've noticed this especially when creating mention lists, e.g.…

```
* @username
* @username
* @username
```

**Comment by charrondev** [comment]
I recently implemented some fixes that should fix mentions in all other editors, and improve things in the WYSIWYG. These fixes will be part of the 2.6 release. Unfortunately the WYSIWYG editor we currently use is dated, abandoned, >10k lines of JS in one file, and is lying to our dated at-mention library about where the cursor is. As a result this issue was not perfectly resolved for that editor.

It is not likely that we will be resolving these issues in the current editor any further, as we are actively working on a new rich editor based on [QuillJS](https://quilljs.com/) for the next upcoming release.

That said I certainly wouldn't turn down a pull request if someone else manages to resolve this issue.

**Comment by charrondev** [comment]
Non-security fixes will no longer be prioritized for advanced editor by the vanilla team. Rich Editor ships with Vanilla 2.8 as the default. https://github.com/vanilla/vanilla/pull/8302
